common-changenotifier/src/main/java/org/duracloud/common/changenotifier/SnsSubscriptionManager.java

```diff
@@ -72,14 +71,13 @@ public synchronized void connect() {
         attributes.put("ReceiveMessageWaitTimeSeconds", "20");
         request.setAttributes(attributes);
         CreateQueueResult result;
-        try {
+
+        this.queueUrl = getQueueUrl(this.queueName);
+        if (this.queueUrl == null) {
```

Comment:
Couldn't the code above, under the "//create sqs queue" comment, move into this if{} block as well? There's only a need to create a CreateQueueRequest once you know you need to create a queue. Also, the log statement log.info("creating sqs queue") will be misleading if a queue doesn't actually need to be created.